docs/website/docs/general-usage/rest-client.md

```diff
@@ -0,0 +1,259 @@
+# Reading data from RESTful APIs
+
+RESTful APIs are a common way to interact with web services. They are based on the HTTP protocol and are used to read and write data from and to a web server. dlt provides a simple way to read data from RESTful APIs using two helper methods: [a wrapper around the `requests`](../reference/performance#using-the-built-in-requests-client) library and a `RESTClient` class.
```

Comment:
I would suggest splitting this in two pages: 
1) RESTClient -- takes care of pagination, authentification, customization
2) requests module (migrate stuff from performance to separate page) -- takes care of retries

Maybe even abstract the requests helper and make retry customization available through RESTClient so users have one entry point.